sips/sip-015/sip-015-network-upgrade.md

```diff
@@ -0,0 +1,1595 @@
+partial stacking operations (i.e., separating the locking operation from
+the stacking operation), which are necessary for supporting this use case.
+Users who wish to do this should use the delegation interfaces.
+
+#### Delegated stackers
+
+Even in PoX-1, delegation operators are capable of assigning multiple
+reward addresses for funds that they manage: each invocation of
+`delegate-stack-stx` can use a different PoX reward address. However,
+if operators wished to split the same _account's_ lockups across
+multiple addresses, this would not be possible.
+
+In PoX-2, however, delegation operators can use
+`delegate-stack-increase` to achieve this. This method allows the
+delegation operator to set a new PoX address to receive the
+"partially stacked" funds from the increase. Once called, the operator
+can use `stack-aggregation-commit` to commit to each PoX reward
+address separately.
+
+## Clarity
+
+If this SIP is ratified, then at the time of the
+Stacks 2.1 network upgrade, the Clarity smart
+contract language will be expanded with new features and some new
+behaviors for existing native methods. In order to support these
+changes, this SIP proposes to introduce *Clarity versioning for smart
+contracts*. In this scheme, each smart contract will be associated
+with a particular Clarity version. The execution environment will
+track the current version for a given execution (in the
+implementation, this will be via the `ContractContext`), and use that
+to select which features are available, and which native method
+implementations will be used. Clarity 2 contracts can invoke Clarity 1
+contracts (and vice-versa), but paticular care will need to be taken
+if a new native keyword is used in the Clarity 1 contract's API.
+
+For example:
+
+```
+Contract A (Clarity 1 Contract):
+  (define-public (stx-account) ...)
+  
+Contract B (Clarity 2 Contract):
+  (contract-call contract-A stx-account)
+```
+
+In such cases, the new keyword (in the above example, the keyword is
+`stx-account`) *cannot* be used by the Clarity 2 contract, and
+therefore invoking the public method of Contract A from Contract B is
+not allowed. To address these cases, contract authors must launch a
+Clarity 1 contract that interposes on Contract A, providing a Clarity
+2 compatible interface.
+
+Existing, pre-2.1 contracts will all be Clarity 1. New contracts will
+default to Clarity 2, and a new Stacks transaction wire format for publishing
+smart contracts will allow contract publishers to choose the version that
+their contract should use.
+
+### New method: `stx-account`
+
+* **Input Signature:** `(stx-account (principal-address principal))`
+* **Output Signature:** `{ locked: uint, unlock-height: uint, unlocked: uint }`
+
+This method returns the status of STX account: the account's
+current STX balance, the amount of STX that is currently locked,
+the unlock height for the account, and whether or not the account
```

Comment:
Ah, I think that was an oversight.  Originally, we thought there would need to be an explicit function to unlock STX.  But since then, Aaron came up with a way for STX tokens automatically unlock if they don't win any reward slots.  This obviated the need for this fourth field, but the SIP text didn't get updated to reflect this.